src/Workspaces/Core/Portable/Workspace/Host/TemporaryStorage/TemporaryStorageHandle.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO.MemoryMappedFiles;
+using System.Threading;
+
+namespace Microsoft.CodeAnalysis.Host;
+
+/// <summary>
+/// Represents a handle to data stored to temporary storage (generally a memory mapped file).  As long as this handle is
```

Comment:
>generally [](http://example.com/codeflow?start=61&length=9)

generally or always?

Reply:
generally.  We have a 'trivial' version that doens't use an MMF.  Though i have no idea nowadays when that would apply.  We used to have it when we had VS4Mac, since we were running on mono.  Nowadays, it's a fallback.  but i don't think we have any real scenarios that use it.